pkg/importer/dataStream.go

```diff
@@ -266,6 +273,110 @@ func (d *DataStream) http() (io.ReadCloser, error) {
 	return countingReader, nil
 }
 
+type registryData struct {
+	dataDir string
+	file    *os.File
+}
+
+//_regData - global variable that stores temporary data of registy flow
+// to be deleted when teh flow ends
+var _regData = registryData{"", nil}
+
+type registryDataCloser struct {
+	io.Reader
+}
+
+func (registryDataCloser) Close() error {
+	_regData.file.Close()
+	os.RemoveAll(_regData.dataDir)
+	return nil
+}
+
+// RegistryDataCloser returns a ReadCloser with a Close method wrapping delettion
+// of temporary data of registry flow
+func RegistryDataCloser(r io.Reader) io.ReadCloser {
+	return registryDataCloser{r}
+}
+
+const (
+	//ContainerDiskImageDir - Expected disk image location in contianer image as described in
+	//https://github.com/kubevirt/kubevirt/blob/master/docs/container-register-disks.md
+	ContainerDiskImageDir = "disk"
+	//TempContainerDiskDir - Temporary location for pulled containerImage
+	TempContainerDiskDir = "tmp"
+)
+
+//This import source downloads specified container image from registry location
+//Then it extracts the image to a temporary location and expects an image file to be located under /disk directory
+//If such exists it creates a Reader on it and returns it for further processing
+func (d *DataStream) registry() (io.ReadCloser, error) {
+	//generate random name for temporary directory
+	_regData.dataDir = filepath.Join(d.DataDir, TempContainerDiskDir)
+
+	imageDir := filepath.Join(_regData.dataDir, ContainerDiskImageDir)
+
+	//create temporary directory if note exists to which all the data will be extracted
+	if _, err := os.Stat(_regData.dataDir); os.IsNotExist(err) {
+		err := os.Mkdir(_regData.dataDir, os.ModeDir|os.ModePerm)
+		if err != nil {
+			return nil, errors.Wrapf(err, fmt.Sprintf("Failed to create tempdirectory %s", _regData.dataDir))
+		}
+	}
+
+	//copy image from registry to the temporary location
+	glog.V(1).Infof("using skopeo to copy from registry")
+	err := image.CopyDirFromRegistryImage(d.Endpoint, _regData.dataDir, ContainerDiskImageDir, d.AccessKey, d.SecKey)
+	if err != nil {
+		os.RemoveAll(_regData.dataDir)
+		return nil, errors.Wrapf(err, fmt.Sprintf("Failed ro read from registry"))
+	}
+
+	//Search for file in /disk directory - if not found - failure
+	imageFile, err := getImageFileName(imageDir)
```

Comment:
Is it our intention that import from registry only supports the `kubevirt` content type?

Reply:
yes